infrastructure/index.ts

```diff
@@ -267,20 +261,65 @@ crawlDirectory(
     webContentsRootPath,
     (filePath: string) => {
         const relativeFilePath = filePath.replace(webContentsRootPath + "/", "");
+
+        // If the file is a .html file that contains a meta refresh redirect, configure the bucket object to be a
+        // webpage redirect to make S3 return a 301 response when the object is requested. This is done to allow
+        // #anchors in URLs to flow across to the destination page, which browsers will do with 301 redirects, but won't
+        // do with meta refresh redirects.
+        // https://docs.aws.amazon.com/AmazonS3/latest/dev/how-to-page-redirect.html
+        const redirect = getMetaRefreshRedirect(filePath);
+
         const contentFile = new aws.s3.BucketObject(
             relativeFilePath,
             {
                 acl: "public-read",
                 key: relativeFilePath,
                 bucket: contentBucket,
-                source: new pulumi.asset.FileAsset(filePath),
-                contentType: getMimeType(filePath) || undefined,
+
+                // Use "/dev/null" for an empty object if this is a redirect, otherwise the actual file path.
+                source: new pulumi.asset.FileAsset(redirect ? "/dev/null" : filePath),
+
+                // Add `contentType` if this isn't a redirect.
+                ...!redirect && { contentType: getMimeType(filePath) },
+
+                // Add `websiteRedirect` if this is a redirect.
+                ...redirect && { websiteRedirect: redirect },
             },
             {
                 parent: contentBucket,
             });
     });
 
+// Returns the redirect URL if filePath is a .html file that contains a meta refresh redirect, otherwise undefined.
+function getMetaRefreshRedirect(filePath: string): string | undefined {
```

Comment:
As a suggestion, I'm not a fan of us checking the `filePath` for the meta refresh redirect, and then also modifying it before returning. That kinda hides/obscures a really important point.

Instead, could we just have `getMetaRefreshRedirect` return just the value from the HTML file and then have a separate `translateFileRedirect` method to do the modification? That makes the code clearer, IMHO. But it isn't that big of a deal.

Reply:
Sounds good. Fixed.